src/main/scala/com/databricks/labs/overwatch/pipeline/Restore.scala

```diff
@@ -0,0 +1,129 @@
+                                ): Unit = {
+    val acceptableCloneLevels = Array("DEEP", "SHALLOW")
+    require(acceptableCloneLevels.contains(cloneLevel.toUpperCase), s"SNAP CLONE ERROR: cloneLevel provided is " +
+      s"$cloneLevel. CloneLevels supported are ${acceptableCloneLevels.mkString(",")}.")
+
+    try {
+      dbutils.fs.ls(targetPrefix)
+    } catch {
+      case e: Exception => dbutils.fs.mkdirs(targetPrefix)
+    }
+
+    try {
+      if (dbutils.fs.ls(targetPrefix).isEmpty) {
+        val cloneSpecs = new Snapshot(workSpace.getConfig.databaseName, targetPrefix, workspace, workspace.database,
+          workspace.getConfig, "restore").buildCloneSpecs(cloneLevel, allTargets)
+        val cloneReport = Helpers.parClone(cloneSpecs)
+        val restoreReportPath = s"${targetPrefix}/restore_report/"
+        logger.log(Level.INFO,s"Restoration is successful. Please check restoreReportPath ${restoreReportPath} for more details")
+        cloneReport.toDS.write.format("delta").mode("append").save(restoreReportPath)
+      } else {
+        logger.log(Level.INFO, "Target Path is not Empty...... Could not proceed with Restoration")
+      }
+
+    } catch {
+      case e: Throwable =>
+        val failMsg = PipelineFunctions.appendStackStrace(e)
+        logger.log(Level.WARN, failMsg)
+    }
+  }
+}
+
+object Restore extends SparkSessionWrapper {
+  private val logger: Logger = Logger.getLogger(this.getClass)
+
+    def apply(
+               workSpace: Workspace,
+               allTarget: Array[PipelineTable],
+               targetPrefix : String,
+               backupPath: String,
+               cloneLevel: String
+             ): Unit = {
+
+      val restoration = new Restore(workSpace.getConfig.databaseName, targetPrefix, workSpace, workSpace.database, workSpace.getConfig)
+      restoration.restore(allTarget,cloneLevel)
+
+    }
+
+    /**
+     * Create a backup of the Overwatch datasets
+     *
+     * @param arg(0)        Source ETL Path Prefix from where restore need to be performed
+     * @param arg(1)        Target ETL Path Prefix to where restore data would be loaded.
+     * @return
+     */
+
+    def main(args: Array[String]): Unit = {
+
+      val sourcePrefix = args(0)
+      val orgID = Initializer.getOrgId
+      val targetPrefix = args(1)
+      val cloneLevel = "Deep"
+      //
+
+
+      val sourcePath = s"${sourcePrefix}/data"
+      val pipReportPath = s"${sourcePath}/pipeline_report"
+      try {
+        dbutils.fs.ls(s"$pipReportPath/_delta_log").nonEmpty
+        logger.log(Level.INFO, s"This ${pipReportPath} location corresponds to previous Overwatch Deployment... proceed Restoration")
+      } catch {
+        case e: FileNotFoundException =>
+          val msg = s"This ${pipReportPath} location does not corresponds to previous Overwatch Deployment...can not proceed with Restoration"
+          logger.log(Level.ERROR, msg)
+          throw new BadConfigException(msg)
+      }
```

Comment:
if we don't have access to the files then how we are handling the exception?

Reply:
Done